### PR TITLE
Validates

### PR DESCRIPTION
**Pull request by megha1302**

### CODE REVIEW

File: app/controllers/orderproducts_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class OrderproductsController < ApplicationController
   before_action :set_orderproduct, only: [:show, :edit, :update, :destroy]
-
+  before_action :required_user
```
Comment (SnehdeepMohite) [inline app/controllers/orderproducts_controller.rb:3]: have you copied this method from someone in your team?

File: app/controllers/orderproducts_controller.rb [hunk 2]
```diff
@@ -67,6 +67,12 @@ def set_orderproduct
       @orderproduct = Orderproduct.find(params[:id])
     end
 
+    def required_user
```
Comment (SnehdeepMohite) [inline app/controllers/orderproducts_controller.rb:70]: what does this line do?

File: app/controllers/ordrs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+class OrdrsController < ApplicationController
+  before_action :set_ordr, only: [:show, :edit, :update, :destroy]
+
+  # GET /ordrs
+  # GET /ordrs.json
+  def index
+    @ordrs = Ordr.all
```
Comment (SnehdeepMohite) [inline app/controllers/ordrs_controller.rb]: Ordr is your Model name?????????????

File: app/models/ordr.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Ordr < ApplicationRecord
```
Comment (SnehdeepMohite) [inline app/models/ordr.rb]: Rename this Model name

update through out the project

File: app/models/product.rb [hunk 5]
```diff
@@ -1,8 +1,8 @@
 class Product < ApplicationRecord
 
-  attr_accessor :orders_attributes, :orderproducts_attributes
-    has_many :orderproducts, dependent: :destroy
-    has_many :orders , through: :orderproducts
+    attr_accessor :ordrs_attributes, :ordrproducts_attributes
+    has_many :ordrproducts, dependent: :destroy
```
Comment (SnehdeepMohite) [inline app/models/product.rb]: Update ordrproducts to order_products

Implement this through out project

**Review by SnehdeepMohite — CHANGES_REQUESTED** [review]

File: app/controllers/orderproducts_controller.rb (hunk 1 above)
Comment (megha1302) [reply app/controllers/orderproducts_controller.rb:3]: No, it was given in udemy videos to implement url level restrictions.I implemented it using refering that video

File: app/controllers/orderproducts_controller.rb (hunk 2 above)
Comment (megha1302) [reply app/controllers/orderproducts_controller.rb:70]: It checks whether the user is logged_in or not 
If user not present it redirects you to root_path 
It restricts crud operations for that controller untill the user is logged in as i used before_action

File: app/controllers/ordrs_controller.rb (hunk 3 above)
Comment (megha1302) [reply app/controllers/ordrs_controller.rb]: Yes , as i mistakenly added orders references to itself and i created a new scaffold with name ordr with the same functionality

**Pull request closed without merging**